hogben/utils.py

```diff
@@ -200,13 +213,22 @@ def fisher(qs, xi, counts, models, step=0.005):
             lb = np.array([param.bounds.limits[0] for param in xi])
             ub = np.array([param.bounds.limits[1] for param in xi])
 
-        # Using the equations from the paper for the coordinate transform.
-        H = np.diag(1/(ub-lb))
-        g = np.dot(np.dot(H.T, g), H)
-
-    # Return the Fisher information matrix.
+        # Scale each parameter with their specified importance,
+        # scale with one if no importance was specified.
+        importance_array = []
+        for param in xi:
+            if hasattr(param, "importance"):
+                importance_array.append(param.importance)
+            else:
+                importance_array.append(1)
+        importance = np.diag(importance_array)
+        H = np.diag(1 / (ub - lb))  # Unit scaling
+        g = np.dot(np.dot(np.dot(H.T, g), H), importance)  # Importance scaling
```

Comment:
probably split this into a couple of calculations as three dot products in a row is a bit much to read